compiler/src/iree/compiler/Dialect/HAL/Conversion/StreamToHAL/Patterns.cpp

```diff
@@ -995,6 +995,49 @@ static void insertSerializationBarriers(Location loc, Block &block,
   }
 }
 
+// Checks if |executeOp| contains only a single transfer operation and returns
+// it. Non-transfer/dispatch operations like cache control will be ignored.
+//
+// Intended to match things like:
+//   stream.cmd.execute ... {
+//     stream.cmd.invalidate
+//     stream.cmd.fill        <----- returned
+//     stream.cmd.flush
+//   }
+// And not:
+//   stream.cmd.execute ... {
+//     stream.cmd.invalidate
+//     stream.cmd.fill
+//     stream.cmd.flush
+//     stream.cmd.dispatch
+//   }
```

Comment:
Nice. This implementation is pretty easy to follow using MLIR :D

Thanks for the comment here too. I find that easier to read than than unit tests.